### PR TITLE
PEP8: bulk fixes on entire grass7/ subdirectory tree with flake8

### PR DESCRIPTION
**Pull request by neteler**
While not being a big fan of bulk fixes I have taken liberty to apply flake8 recursively on `grass-addons/grass7/`.

The fixes are separated by commit in order to more easily review them.
Eventually, after review, all can be squashed into one commit.

Rationale: follow more closely our coding style: https://trac.osgeo.org/grass/wiki/Submitting/Python#PEP8standardStyle

Once accepted, the [GitHub Action flake8 test](https://github.com/OSGeo/grass/blob/master/.github/workflows/flake8.yml) from  core GRASS GIS may be used also here.

The script I wrote to auto-create this PR:

```
cat pep8_formatting_commit.sh 
#!/bin/sh

# requires "python3-autopep8"

# put desired error codes to be fixed into CSV file
# https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes
#
# format:
# E117 over-indented
# E122 continuation line missing indentation or outdented
# ...

CODES=~/pep8_formatting_codes_addons.csv

# trick: using a non-english char as a special separator
for i in $(cat $CODES | sed 's+ +ä+g') ; do
    code="$(echo $i | cut -d'ä' -f1)"
    desc=$(echo $i | cut -d'ä' -f2- | sed 's+ä+ +g')
    echo "Processing: $code: $desc..."
    ag --python -l | xargs autopep8 --select $code --aggressive --aggressive --in-place --jobs 4
    LISTTOCOMMIT=$(git status | grep modified | tr -s ' ' ' ' | cut -d' ' -f2)
    git commit -m"PEP8: fix $code: $desc" $LISTTOCOMMIT
done
```

### CODE REVIEW

**Comment by neteler** [comment]
To simplify the review, below the non-trivial changes:

(use mouse-over to see the description)
- https://github.com/OSGeo/grass-addons/pull/357/commits/65ce2b697c7539ae9fcdb645ee0bd807e06c47e7
- https://github.com/OSGeo/grass-addons/pull/357/commits/23b9baf515c8d3b436e1a81dc9d408cd17114132
- https://github.com/OSGeo/grass-addons/pull/357/commits/755cf54a183340934ac5efb61a2e82539ada7494
- https://github.com/OSGeo/grass-addons/pull/357/commits/fa2b282d02067ac967c01702b30958635588dcf7
- https://github.com/OSGeo/grass-addons/pull/357/commits/0c4f642d81bf9812ee3c102bf79a24899fe2c7a7
- https://github.com/OSGeo/grass-addons/pull/357/commits/9abbd58980bc802876c632ba99afa5332836ba25
- https://github.com/OSGeo/grass-addons/pull/357/commits/bf103a5a0461e1928419f6b542fca589f6858455

Leftover python debugger?
- https://github.com/OSGeo/grass-addons/pull/357/commits/7e5fde8cf99527e235b158cfa92822c397cbce07

The other commits in this PR are wrong blanks, semicolons, mixed tab/white space, and such being cleaned up.

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
Thanks, I checked what you linked. In the following, a manual change would be better:

```
# grass7/gui/wxpython/wx.rdigit/rdigit/dialogs_core.py
-   UsingGDAL  = vExternalOut.has_key('Not using GDAL') == False
+   UsingGDAL  = "Not using GDAL" not in vExternalOut
```

Although that looks like super fragile code in any case.

You are right about `import ipdb; ipdb.set_trace()`. It should be just deleted.

If I remember correctly, bulk autopep run on wxGUI code broke several places mainly due to bad replacement of tabs by spaces (perhaps autopep did not expect tab to be used for character number reduction). It would be good to check files which had tabs on lines with some code (the relevant commit has many files with tabs only at empty lines).

**Comment by neteler** [comment]
Thanks @wenzeslaus - changed as suggested.

**Comment by hellik** [comment]
scrolled through my addons and a few others where I've been involved (e.g. r.basin, r.hazard.flood, r.hypso, ...), looks good so far

**Comment by hellik** [comment]
from the submitting guidelines:

```
    Do not use space around parentheses: 

# use this:
grass.run_command('g.region', raster='myrast')
# not this:
grass.run_command( 'g.region', raster='myrast' )
```
though in the bulk refactoring only the first space around the first parentheses is deleted, the second is still there.

for example:

```
grass.message( "Starting importing GBIF data ..." ) 
```

changed in

```
grass.message("Starting importing GBIF data ..." )
```

**Comment by neteler** [comment]
> though in the bulk refactoring only the first space around the first parentheses is deleted, the second is still there.

Indeed. While this rule here https://www.flake8rules.com/rules/E201.html states

_"Open parentheses should not have any space before or after them."_

(which was in this PR done with https://github.com/OSGeo/grass-addons/pull/357/commits/1205fa609ba6ab8a27b1fc629c9966e63ba67655)

I do not find the corresponding "Closing parentheses ..." pendant for it.

Does anyone know the flake8 code to be applied here?

**Comment by neteler** [comment]
... found it :-)

_E202: whitespace before ‘)’_

**Comment by wenzeslaus** [comment]
1. The whitespace would be ultimately best handled by Black I think, but this will get it closer at least for single line statements.
2. The E202 issue surprised me. I thought you are actually running flake8 on it afterwards, but it does not seem so.
3. Any idea about list of files which had a tab at a non-whitespace line? These would be the ones to check manually (see above).

**Comment by neteler** [comment]
>     1. The whitespace would be ultimately best handled by Black I think, but this will get it closer at least for single line statements.

I suggest to separate Black application out into a new PR. This PR is already large enough.

>     2. The E202 issue surprised me. I thought you are actually running flake8 on it afterwards, but it does not seem so.

Not sure what you mean. I simply overlooked to apply E202 initially when preparing this PR.
 
>     3. Any idea about list of files which had a tab at a non-whitespace line? These would be the ones to check manually (see above).

I checked and identified these related commits:
(use mouse-over to see the description)
- https://github.com/OSGeo/grass-addons/pull/357/commits/2ac5f0f987da2c4d449948046c53d2c194db94b6
- https://github.com/OSGeo/grass-addons/pull/357/commits/46e7451eadb3ab96114e5d9ad193c0170811a563
- https://github.com/OSGeo/grass-addons/pull/357/commits/3e7ea28587c8ddc262ab5afd7a7f8b1eb31d9ee3
- https://github.com/OSGeo/grass-addons/pull/357/commits/1bfa3b5a777e48c5f5dc53d3ab0530224a48986c
- https://github.com/OSGeo/grass-addons/pull/357/commits/876bec72d9f112752b407bff409e98e221b1089b

File: grass7/imagery/i.landsat8.swlst/column_water_vapor.py [hunk 1]
```diff
@@ -371,16 +371,16 @@ def _big_cwv_expression(self):
...
-               '\ \n'
-               '\ \n  tj_mean = {tjm},'
-               '\ \n'
-               '\ \n  numerator = {numerator},'
-               '\ \n'
-               '\ \n  denominator = {denominator},'
-               '\ \n'
-               '\ \n  rji = numerator / denominator,'
-               '\ \n'
+               r'\ \n  ti_mean = {tim},'
```
Comment (ninsbl) [inline grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: Does the expression still work with the change to raw string?

Comment (neteler) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: Does @NikosAlexandris have an idea here?

**Comment by wenzeslaus** [comment]
> >
> > 1. The whitespace would be ultimately best handled by Black I think, but this will get it closer at least for single line statements.
> >
> 
> I suggest to separate Black application out into a new PR. This PR is already large enough.

I agree. Just saying the Black whitespace is different than the Flake8 (still PEP8/Flake8 complaint, but more strict and sometimes different than other auto-formatting tools.

> >
> > 2. The E202 issue surprised me. I thought you are actually running flake8 on it afterwards, but it does not seem so.
> >
> 
> Not sure what you mean. I simply overlooked to apply E202 initially when preparing this PR.

I mean running the Flake8 checks. It seemed to me you are talking about auto-applying changes.

> >
> > 3. Any idea about list of files which had a tab at a non-whitespace line? These would be the ones to check manually (see above).
> >
> 
> I checked and identified these related commits:
> 
>  * [2ac5f0f](https://github.com/OSGeo/grass-addons/commit/2ac5f0f987da2c4d449948046c53d2c194db94b6)
>  * [46e7451](https://github.com/OSGeo/grass-addons/commit/46e7451eadb3ab96114e5d9ad193c0170811a563)
>  * [3e7ea28](https://github.com/OSGeo/grass-addons/commit/3e7ea28587c8ddc262ab5afd7a7f8b1eb31d9ee3)
>  * [1bfa3b5](https://github.com/OSGeo/grass-addons/commit/1bfa3b5a777e48c5f5dc53d3ab0530224a48986c)
>  * [876bec7](https://github.com/OSGeo/grass-addons/commit/876bec72d9f112752b407bff409e98e221b1089b)

These look good. No changes in if branches.

File: grass7/imagery/i.landsat8.swlst/column_water_vapor.py (hunk 1 above)
Comment (neteler) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: Shall I revert this part? I really want to merge this PR.

Comment (ninsbl) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: I just ran a little test. Conversion to raw strings seem to break the mapcalculator expressions:

```
import grass.script as gscript

expression = 'mc_test=1*2+\ \n3' 
expression_r = r'mc_testr=1*2+\ \n3' 

# This works
gscript.mapcalc(expression)
```

However, this:
```
gscript.mapcalc(expression_r)

```
throws the following error message:
```
Invalid map <\>
Parse error
ERROR: parse error
ERROR: An error occurred while running r.mapcalc with expression:
       mc_testr=1*2+\ \n3
ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.
```
(...)

So, yes, I am afraid the part where strings are converted to raw strings should be reverted...

Comment (NikosAlexandris) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: > Shall I revert this part? I really want to merge this PR.

I didn't the time to look if raw strings would make any difference.
I can check upon this tonight.

Comment (neteler) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py:374]: @NikosAlexandris did you have a chance to check?

Comment (neteler) [reply grass7/imagery/i.landsat8.swlst/column_water_vapor.py]: Reverted in 1d1594392ea1afadf1d1a9082c5faa70c6637ab6

Thanks for having spotted this issue.

**Comment by neteler** [comment]
Is there anything else left open or may I merge?

**Pull request merged**